src/ert/_c_wrappers/enkf/config/field_config.py

```diff
@@ -66,21 +66,28 @@ def save(self, run_path: Path, real_nr: int, ensemble: EnsembleReader):
             self.truncation_min,
             self.truncation_max,
         )
-
+        data = np.ma.MaskedArray(
+            data=data, mask=self.mask, fill_value=np.nan
+        )  # type: ignore
         field_utils.save_field(
             data,
             self.name,
-            ensemble.experiment.grid_path,
-            field_utils.Shape(self.nx, self.ny, self.nz),
             file_out,
             self.file_format,
         )
+
         _logger.debug(f"save() time_used {(time.perf_counter() - t):.4f}s")
 
     def save_experiment_data(self, experiment_path):
-        grid_filename = "grid" + Path(self.grid_file).suffix.upper()
-        if not (experiment_path / grid_filename).exists():
-            shutil.copy(self.grid_file, experiment_path / grid_filename)
+        mask_path = experiment_path / f"{Path(self.grid_file).stem}.npy"
```

Comment:
As long as there is only one possible `GRID` there should only be a single mask file with a constant name, eg. `grid_mask`.